packages/kbn-esql-utils/src/utils/query_parsing_helpers.ts

```diff
@@ -25,8 +26,18 @@ export function getIndexPatternFromSQLQuery(sqlQuery?: string): string {
   return '';
 }
 
+// retrieves the index pattern from the aggregate query for ES|QL using ast parsing
+export async function getIndexPatternFromESQLQuery(esql?: string) {
+  const { ast } = await getAstAndSyntaxErrors(esql);
+  const fromCommand = ast.find(({ name }) => name === 'from');
+  const args = fromCommand?.args as ESQLSource[];
+  const indices = args.filter((arg) => arg.sourceType === 'index');
+  return indices?.map((index) => index.text).join(',');
+}
+
 // retrieves the index pattern from the aggregate query for ES|QL
-export function getIndexPatternFromESQLQuery(esql?: string): string {
+// this can fail if from is mentioned in comments etc
+export function getIndexPatternFromESQLQueryDeprecated(esql?: string): string {
```

Comment:
ℹ️ Named this deprecated as there are still usages, waiting from the consumers to fix as they know their apps better than me.